include/charmlite/core/destination.hpp

```diff
@@ -22,6 +22,7 @@ namespace cmk {
             // reserved for collection communication
             // TODO ( so rename it as such! )
             bcast_id_t bcast;
+            int pe;
```

Comment:
In the model of Converse, messages are only assigned a PE at the callsite of `send`. We can overload the function with an alternative that enforces a specific destination PE, and avoid adding this to the header. `insert` can use this to route it directly to the desired PE.

In Charm++, there are two types of array element creation messages: `createhere` or `createhome`. If a PE receives a `createhere` message, it creates it regardless of whether it's its home. We can replicate this behavior here with a boolean flag in the message.